packages/lib/src/vis/line/GlyphMaterial.tsx

```diff
@@ -1,29 +1,96 @@
-import { useLoader } from '@react-three/fiber';
-import { NearestFilter, TextureLoader } from 'three';
+import { Color } from 'three';
 
-import { GLYPH_URLS } from './models';
+import { GlyphType } from './models';
 
 interface Props {
   color?: string;
   size: number;
-  glyphURL?: string;
-  vertexColors?: boolean;
+  glyphType?: GlyphType;
 }
 
+// From https://github.com/silx-kit/silx/blob/11f764176ede6ad68d7551e215773a47bcb4614d/src/silx/gui/plot/backends/glutils/GLPlotCurve.py#L581-L799
+const GLYPH_SHADERS = {
+  [GlyphType.Cross]: `
+    float alphaSymbol(vec2 coord, float size) {
+      vec2 pos = floor(size * coord) + 0.5;
+      vec2 d_x = abs(pos.x + vec2(- pos.y, pos.y - size));
+      if (min(d_x.x, d_x.y) <= 0.5) {
+          return 1.0;
+      } else {
+          return 0.0;
+      }
+    }
+  `,
+  [GlyphType.Cap]: `
+    float alphaSymbol(vec2 coord, float size) {
+      float dy = abs(size * (coord.y - 0.5));
+      if (dy < 0.5) {
+          return 1.0;
+      } else {
+          return 0.0;
+      }
+    }
+  `,
+  [GlyphType.Circle]: `
+    float alphaSymbol(vec2 coord, float size) {
+      float radius = 0.5;
+      float r = distance(coord, vec2(0.5, 0.5));
+      return clamp(size * (radius - r), 0.0, 1.0);
+    }
+  `,
+  [GlyphType.Square]: `
+    float alphaSymbol(vec2 coord, float size) {
```

Comment:
Do the arguments have to be specified even though they're not used?

Reply:
Yes. The call in the fragment shader will still pass the two arguments so all `alphaSymbol` functions must have the same signature.

GLSL makes no compromise :joy: